plugin/src/utils/initial_scarb_codes.ts

```diff
@@ -2,9 +2,16 @@ import axios from 'axios'
 
 export async function fetchGitHubFilesRecursively (
   repository: string,
-  path: string
+  path: string,
+  ref?: string
 ): Promise<Array<({ fileName: string, content: string, path: string } | null)>> {
-  const apiUrl = `https://api.github.com/repos/${repository}/contents/${path}`
+  const apiUrl = (function () {
```

Comment:
Curious why this needs to be IIFE ?
can't we just do this
```
const apiUrl  = `https://api.github.com/repos/${repository}/contents/${path}${ref === undefined ? '' : `?ref=${ref}`}`

Reply:
Just a preference. Can be anything. But I would leave it like this, other option looks a bit clumsy for me.